### PR TITLE
[Horizon] Tag component

### PR DESCRIPTION
**Pull request by szabinst**
Adds the Tag component.

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Horizon Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Build failed, skipping coverage check</td>
    </tr>
  </tbody>
</table>



<details><summary>:x: XCTest failed: CoreTests/AssignmentCellViewModelTests/testSubmissionStatusAndIconAndColor</summary>


    XCTAssertEqual failed: ("#6a7883") is not equal to ("#697783")
    XCTAssertEqual failed: ("#03893d") is not equal to ("#03893c")
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 145f279ae37fbef6d90bff92f8e9a823ecbc51fe
</p>

File: packages/HorizonUI/Sources/HorizonUI/Sources/Components/Tag/HorizonUI.Tag.swift [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        private let onCloseAction: (() -> Void)?
+
+        // MARK: - Init
+
+        init(
+            title: String,
+            style: Style,
+            size: Size,
+            backgroundColor: Color = .huiColors.surface.pageSecondary,
+            borderColor: Color = .huiColors.lineAndBorders.lineStroke,
```
Comment (Ahmed-Naguib93) [inline packages/HorizonUI/Sources/HorizonUI/Sources/Components/Tag/HorizonUI.Tag.swift:67]: The `backgroundColor` is added but not used anywhere, & i recommend to pass text color too as a parameter

**Review by Ahmed-Naguib93 — CHANGES_REQUESTED** [review]
Thanks for your work, 
Is working fine for me, i add one comment

File: packages/HorizonUI/Sources/HorizonUI/Sources/Components/Tag/HorizonUI.Tag.swift (hunk 1 above)
Comment (szabinst) [reply packages/HorizonUI/Sources/HorizonUI/Sources/Components/Tag/HorizonUI.Tag.swift:70]: Good catch. I added text color and also disabled state was missing.

**Review by Ahmed-Naguib93 — APPROVED** [review]

**Pull request merged**